src/HotCorners/HotCorner.vala

```diff
@@ -34,6 +34,20 @@ public class Gala.HotCorner : Object {
     public const string POSITION_BOTTOM_RIGHT = "hotcorner-bottomright";
     private const int BARRIER_SIZE = 30;
 
+    /**
+     * Gets mouse pointer speed, a value between -1 and 1.
+     * uses this setting to create a scaling factor only for values above 0, 
+     * since smaller values are less sensitive and don't need adjusted
+     */
+    private static GLib.Settings mouse_settings;
+    private static double mouse_speed_scale_factor;
+    static construct {
+        mouse_settings = new GLib.Settings ("org.gnome.desktop.peripherals.mouse");
+        mouse_speed_scale_factor = (mouse_settings.get_double("speed") > 0) ? 
+            (1 + mouse_settings.get_double("speed")) * 2 :
+            1;
```

Comment:
I think that this part should be in a `construct` block as we will also need to watch for the change of the `speed` property

Reply:
Thanks for that! I was wondering how to do that.